src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -81,6 +85,40 @@ ss::future<> service::do_start() {
     }
 }
 
+ss::future<> service::create_internal_topic() {
```

Comment:
is this the action under control of `one_shot`? seems like something we could do when schema registry starts up?

Reply:
> is this the action under control of `one_shot`? seems like something we could do when schema registry starts up?

Yes, it's under control of one_shot. The idea was that if the schema registry is never used, there's no point in consuming a bunch of memory loading the schemas and connecting the client to each broker. It ties in with some of the previous suggestions of making the client broker connections lazy.

Reply:
makes sense. we can merge this and fix it up later, but a couple observations:

1. internal topics like this really don't cost anything to keep around. so if we started redpanda with the schema registry enabled but no one ever used it the cost would be some empty files. we've discussed in the past creating some internal topics like this at cluster creation time.

2. I wonder if you could create a server where every handler creates the topic, and then returns 503 or something and you immediately swap in a different server implementation with the real handlers that assume the topic exists.

Reply:
There's still the question of when to fetch the topic into the schema store, which is also covered by one_shot, and has potentially significant memory usage.

At some point it will make sense to create an index in memory so that schemas can be evicted and retrieved at will, but for now, everything is in memory.

Reply:
I like the idea, though, and I guess it's equally applicable.